### PR TITLE
First draft of changes

### PR DESCRIPTION
**Pull request by kalyankrishna1**
Updates sample to the current set of libraries and some more

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,18 +30,26 @@ From your shell or command line:
 
 ### Step 2:  Register the sample with your Azure Active Directory tenant
 
-There are two projects in this sample.  Each needs to be separately registered in your Azure AD tenant.
+There are two projects in this sample. Each needs to be separately registered in your Azure AD tenant. To register these projects, you can:
+
+- either follow the steps in the paragraphs below ([Step 2](#step-2--register-the-sample-with-your-azure-active-directory-tenant) and [Step 3](#step-3--configure-the-sample-to-use-your-azure-ad-tenant))
```
Comment (andretms) [inline README.md]: - Follow the steps in the ([Step 2](#step-2--register-the-sample-with-your-azure-active-directory-tenant) and [Step 3](#step-3--configure-the-sample-to-use-your-azure-ad-tenant)) to register an application and configure the sample using Azure Portal

File: README.md [hunk 2]
```diff
@@ -30,18 +30,26 @@ From your shell or command line:
 
 ### Step 2:  Register the sample with your Azure Active Directory tenant
 
-There are two projects in this sample.  Each needs to be separately registered in your Azure AD tenant.
+There are two projects in this sample. Each needs to be separately registered in your Azure AD tenant. To register these projects, you can:
+
+- either follow the steps in the paragraphs below ([Step 2](#step-2--register-the-sample-with-your-azure-active-directory-tenant) and [Step 3](#step-3--configure-the-sample-to-use-your-azure-ad-tenant))
+- or use PowerShell scripts that:
+  - **automatically** create for you the Azure AD applications and related objects (passwords, permissions, dependencies)
```
Comment (andretms) [inline README.md]: -- or --

- Follow [these instructions](./AppCreationScripts/AppCreationScripts.md) to run a script that automatically register the Azure AD applications and related objects (passwords, permissions, dependencies) required by this sample and modify the Visual Studio projects' configuration files with the necessary configuration.

File: README.md [hunk 3]
```diff
@@ -30,18 +30,26 @@ From your shell or command line:
...
+
 
 #### Register the TodoListService web API
 
 1. Sign in to the [Azure portal](https://portal.azure.com).
 2. On the top bar, click on your account and under the **Directory** list, choose the Active Directory tenant where you wish to register your application.
 3. Click on **More Services** in the left hand nav, and choose **Azure Active Directory**.
 4. Click on **App registrations** and choose **Add**.
-5. Enter a friendly name for the application, for example 'TodoListService' and select 'Web Application and/or Web API' as the Application Type. For the sign-on URL, enter the base URL for the sample, which is by default `https://localhost:44321`. Click on **Create** to create the application.
+5. Enter a friendly name for the application, for example 'TodoListService-AppIdentity' and select 'Web Application and/or Web API' as the Application Type. For the sign-on URL, enter the base URL for the sample, which is by default `https://localhost:44321`. Click on **Create** to create the application.
```
Comment (andretms) [inline README.md]: Enter a friendly name for the application, for example `TodoListService-AppIdentity` and in **Application Type**, select **Web Application and/or Web API**. For the sign-on URL, enter the base URL for the sample, which is by default `https://localhost:44321`. Click on **Create** to create the application.

File: README.md [hunk 4]
```diff
@@ -17,7 +17,7 @@ For more information about how the protocols work in this scenario and other sce
 ## How To Run This Sample
 
 To run this sample you will need:
-- Visual Studio 2013
+-  [Visual Studio 2017](https://aka.ms/vsdownload)
 - An Internet connection
 - An Azure Active Directory (Azure AD) tenant. For more information on how to get an Azure AD tenant, please see [How to get an Azure AD tenant](https://azure.microsoft.com/en-us/documentation/articles/active-directory-howto-tenant/) 
 - A user account in your Azure AD tenant. This sample will not work with a Microsoft account, so if you signed in to the Azure portal with a Microsoft account and have never created a user account in your directory before, you need to do that now.
```
Comment (andretms) [inline README.md]: An account in your Azure AD tenant (work account). This sample will accept sign-ins of a personal Microsoft account (such as outlook.com or hotmail.com) unless they these accounts are added as *guests* of your Azure AD tenant.

File: README.md [hunk 5]
```diff
@@ -30,18 +30,26 @@ From your shell or command line:
...
 #### Register the TodoListService web API
 
 1. Sign in to the [Azure portal](https://portal.azure.com).
 2. On the top bar, click on your account and under the **Directory** list, choose the Active Directory tenant where you wish to register your application.
 3. Click on **More Services** in the left hand nav, and choose **Azure Active Directory**.
 4. Click on **App registrations** and choose **Add**.
-5. Enter a friendly name for the application, for example 'TodoListService' and select 'Web Application and/or Web API' as the Application Type. For the sign-on URL, enter the base URL for the sample, which is by default `https://localhost:44321`. Click on **Create** to create the application.
+5. Enter a friendly name for the application, for example 'TodoListService-AppIdentity' and select 'Web Application and/or Web API' as the Application Type. For the sign-on URL, enter the base URL for the sample, which is by default `https://localhost:44321`. Click on **Create** to create the application.
 6. While still in the Azure portal, choose your application, click on **Settings** and choose **Properties**.
 7. Find the Application ID value and copy it to the clipboard.
```
Comment (andretms) [inline README.md]: This seems to be an orphan step, or at least is not in a good sequence

File: README.md [hunk 6]
```diff
@@ -53,7 +61,7 @@ There are two projects in this sample.  Each needs to be separately registered i
 6. While still in the Azure portal, choose your application, click on **Settings** and choose **Properties**.
 7. Find the Application ID value and copy it to the clipboard.
```
Comment (andretms) [inline README.md]: Same thing here: why user is copying the App Id to the clipboard at this point?

**Comment by jmprieur** [comment]
thanks @kalyankrishna1. I'll review the changes this week

**Pull request merged**